exercises/nodejs/src/game.js

```diff
@@ -1,172 +1,226 @@
-    if (currentCategory() == 'Sports') { console.log(sportsQuestions.shift()) }
-    if (currentCategory() == 'Rock') { console.log(rockQuestions.shift()) }
-  }
+    if (this.getInPenaltyBox()) {
+      if (roll % 2 !== 0) {
+        isGettingOutOfPenaltyBox = true;
 
-  this.roll = function (roll) {
-    console.log(players[currentPlayer] + ' is the current player')
-    console.log('They have rolled a ' + roll)
-
-    if (inPenaltyBox[currentPlayer]) {
-      if (roll % 2 != 0) {
-        isGettingOutOfPenaltyBox = true
-
-        console.log(players[currentPlayer] + ' is getting out of the penalty box')
-        places[currentPlayer] = places[currentPlayer] + roll
-        if (places[currentPlayer] > 11) {
-          places[currentPlayer] = places[currentPlayer] - 12
-        }
-
-        console.log(players[currentPlayer] + "'s new location is " + places[currentPlayer])
-        console.log('The category is ' + currentCategory())
-        askQuestion()
+        console.log(
+          players[currentPlayer] + " is getting out of the penalty box"
+        );
+
+        movePlayer(roll);
       } else {
-        console.log(players[currentPlayer] + ' is not getting out of the penalty box')
-        isGettingOutOfPenaltyBox = false
+        console.log(
+          players[currentPlayer] + " is not getting out of the penalty box"
+        );
+        isGettingOutOfPenaltyBox = false;
       }
     } else {
-      places[currentPlayer] = places[currentPlayer] + roll
-      if (places[currentPlayer] > 11) {
-        places[currentPlayer] = places[currentPlayer] - 12
-      }
+      movePlayer(roll);
+    }
+  };
 
-      console.log(players[currentPlayer] + "'s new location is " + places[currentPlayer])
-      console.log('The category is ' + currentCategory())
-      askQuestion()
+  const nextPlayer = () => {
+    currentPlayer += 1;
+    if (currentPlayer === players.length) {
+      currentPlayer = 0;
     }
-  }
+    return true;
+  };
+
+  this.canAnswerQuestion = () =>
+    !this.getInPenaltyBox() || isGettingOutOfPenaltyBox;
 
   this.wasCorrectlyAnswered = function () {
-    if (inPenaltyBox[currentPlayer]) {
-      if (isGettingOutOfPenaltyBox) {
-        console.log('Answer was correct!!!!')
-        purses[currentPlayer] += 1
-        console.log(players[currentPlayer] + ' now has ' +
-            purses[currentPlayer] + ' Gold Coins.')
-
-        var winner = didPlayerWin()
-        currentPlayer += 1
-        if (currentPlayer == players.length) { currentPlayer = 0 }
-
-        return winner
-      } else {
-        currentPlayer += 1
-        if (currentPlayer == players.length) { currentPlayer = 0 }
-        return true
+    const addGoldCoin = () => {
+      console.log("Answer was correct!!!!");
+      purses[currentPlayer] += 1;
+      console.log(
+        getCurrentPlayer() +
+          " now has " +
+          purses[currentPlayer] +
+          " Gold Coins."
+      );
+
+      var winner = checkPlayerHasWon();
```

Comment:
could be worth updating this variable from var to let or const along with the others.